src/webui/gitutils.jl

```diff
@@ -160,12 +161,18 @@ function gettreesha(
             dest = joinpath(dir, r.name)
             run(`git clone $url $dest`)
             run(`git -C $dest checkout $ref`)
-            match(r"tree (.*)", readchomp(`git -C $dest show --format=raw`))[1]
+
+            if isempty(subdir)
+                match(r"tree (.*)", readchomp(`git -C $dest show --format=raw`))[1], ""
+            elseif isdir(joinpath(dest, subdir))
+                match(r"tree (.*)\t", readchomp(`git -C $dest ls-tree -d HEAD $subdir`))[1], ""
+            else
+                nothing, "The sub-directory $subdir does not exist in this repository"
+            end
```

Comment:
This looks fine but if desired all git operations after `git clone` could be simplified to reading the output of `git rev-parse $ref:$subdir`, which will return only the tree hash both when `subdir` is empty and non-empty.